buildlib/tools/zombie_killer.sh

```diff
@@ -1,20 +1,20 @@
-#!/bin/bash -eEx
+#!/bin/bash -eE
 
-# Search for zombie or defunct processes; kill them all
-out=$(ps -eo pid,stat | awk '$2 == "Z|defunct" {print $1}')
+# Search for zombie with high CPU usage; kill them all
+out=$(ps -eo pid,stat,%cpu | grep Z | awk '$3>85.0 {print $1}')
 if [ -z "$out" ]; then
-  echo "No zombies found"
+  echo "No high CPU consuming zombies were found"
   exit 0
 else
   for i in $out ; do
     kill -9 "$i"
   done
 fi
 
-# Search again; tell parents you killed their kids; kill parents
-out=$(ps -eo pid,stat | awk '$2 == "Z|defunct" {print $1}')
+# Search for survivors; tell parents you killed their kids; kill parents
+out=$(ps -eo pid,stat,%cpu | grep Z | awk '$3>85.0 {print $1}')
 for i in $out ; do
-  pid_parent=$(ps -o ppid=$i)
+  pid_parent=$(ps -p "$i" -o ppid=)
   kill -s SIGCHLD $pid_parent
   kill -9 $pid_parent
```

Comment:
Why do we need to terminate parent process with SIGKILL? As you mentioned parent process may be a docker process

Reply:
If a zombie couldn't be dealt with directly, approaching its parents is the only choice left.
For the high CPU-eating zombies it well worth the hustle.

Reply:
But if we parent for example `init` or another important process?

Reply:
It appears we cannot terminate init even with SIGKILL.